src/flags.rs

```diff
@@ -111,6 +112,10 @@ where
     fn configure_from(matches: &ArgMatches, config: &Config) -> T {
         let mut result: T = Default::default();
 
+        if let Some(value) = Self::from_environment() {
```

Comment:
We should be able to invert the order and do an early return for all these cases.